aeron-driver/src/main/c/aeron_publication_image.c

```diff
@@ -1035,11 +1035,8 @@ void aeron_publication_image_check_untethered_subscriptions(
     {
         aeron_tetherable_position_t *tetherable_position = &subscribable->array[i];
 
-        if (tetherable_position->is_tether)
-        {
-            int64_t position = aeron_counter_get_acquire(tetherable_position->value_addr);
-            max_sub_pos = position > max_sub_pos ? position : max_sub_pos;
-        }
+        int64_t position = aeron_counter_get_acquire(tetherable_position->value_addr);
```

Comment:
I think this change is partially correct, but if `!tetherable_position->is_tether` then we need to check the tethered state and exclude LINGERING untethered subscriptions to match with the Java implementation.

Reply:
I couldn't see where java considered RESTING in this calculation. I can see where its excluded afterwards when the subscriptions are iterated on again and state checked (as they are in C) but not in the java method which seems to calculate this position unless I'm missing something? https://github.com/aeron-io/aeron/blob/master/aeron-driver/src/main/java/io/aeron/driver/PublicationImage.java#L1138

Reply:
They get removed from the list of `SubscriberPosition`s: https://github.com/aeron-io/aeron/blob/master/aeron-driver/src/main/java/io/aeron/driver/PublicationImage.java#L1113

Reply:
Ah yes thanks and we can't simply remove from subscription list here because (among other things) it will prevent RESTING state from transitioning to ACTIVE. OK will update the change today.

Reply:
@fquinner The condition is wrong, i.e. we need to check for the `RESTING` state if subscription is untethered:
```c
if (tetherable_position->is_tether || AERON_SUBSCRIPTION_TETHER_RESTING != tetherable_position->state)
```

Reply:
D'oh! OK thanks pushed in an update there now.